sdk/exportentities/workflow_notif.go

```diff
@@ -0,0 +1,208 @@
+package exportentities
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/ovh/cds/sdk"
+)
+
+// Booleans
+var (
+	True  = true
+	False = false
+)
+
+// NotificationEntry represents a notification set on a nodeEntry
+type NotificationEntry struct {
+	Type     string                        `json:"type" yaml:"type"`
+	Settings *sdk.UserNotificationSettings `json:"settings,omitempty" yaml:"settings,omitempty"`
+}
+
+// craftNotificationEntry returns the NotificationEntry and the name of the nodeEntries concerned
+func craftNotificationEntry(w sdk.Workflow, notif sdk.WorkflowNotification) ([]string, NotificationEntry, error) {
+	entry := NotificationEntry{}
+	nodeNames := make([]string, len(notif.SourceNodeIDs))
+	for i, id := range notif.SourceNodeIDs {
+		node := w.GetNode(id)
+		if node == nil {
+			return nil, entry, fmt.Errorf("workflow node %d not found", id)
+		}
+		nodeNames[i] = node.Name
+	}
+	sort.Strings(nodeNames)
+	entry.Type = notif.Type
+	entry.Settings = &notif.Settings
+
+	// Replace the default values by nil
+	if entry.Settings.OnStart != nil && !*entry.Settings.OnStart {
+		entry.Settings.OnStart = nil
+	}
+	if entry.Settings.SendToGroups != nil && !*entry.Settings.SendToGroups {
+		entry.Settings.SendToGroups = nil
+	}
+	if entry.Settings.SendToAuthor != nil && *entry.Settings.SendToAuthor {
+		entry.Settings.SendToAuthor = nil
+	}
+	// Replace the default values by empty strings
+	if entry.Settings.OnSuccess == sdk.UserNotificationChange {
+		entry.Settings.OnSuccess = ""
+	}
+	if entry.Settings.OnFailure == sdk.UserNotificationAlways {
+		entry.Settings.OnFailure = ""
+	}
+	// Replace default templates by nil if they are default values
+	if entry.Settings.Template != nil {
+		defaultTemplate, has := sdk.UserNotificationTemplateMap[entry.Type]
+		if !has {
+			return nil, entry, fmt.Errorf("workflow notification %s not found", entry.Type)
+		}
+		if defaultTemplate.Subject == entry.Settings.Template.Subject {
+			entry.Settings.Template.Subject = ""
+		}
+		if defaultTemplate.Body == entry.Settings.Template.Body {
+			entry.Settings.Template.Body = ""
+		}
+		if entry.Settings.Template.Body == "" && entry.Settings.Template.Subject == "" {
+			entry.Settings.Template = nil
+		}
+	}
+
+	// Finally if settings are all default, lets skip it
+	if entry.Settings.OnFailure == "" &&
```

Comment:
do you think it is possible to compare with an empty struct ?

Reply:
`invalid operation: (struct containing []string cannot be compared)`